### PR TITLE
add sandbox sunset adr

### PR DESCRIPTION
**Pull request by mslarae13**

### CODE REVIEW

File: decisions/0010-sunsetting-sandbox.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+status: accepted
+date: 2024-08-14
+deciders: Montana Smith, NMDC PIs, Mike Nagler, Patrick Kalita 
+consulted: Entire NMDC team via [GitHub discussion](https://github.com/microbiomedata/issues/discussions/743), meetings, and slack were given the oppotrunity to voice opinions
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:5]:
```suggestion
consulted: All NMDC team members—via [GitHub discussion](https://github.com/microbiomedata/issues/discussions/743), meetings, and Slack—were given the opportunity to express their opinions
```

File: decisions/0010-sunsetting-sandbox.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+---
+# Sunsetting the NMDC Sandbox UI
+
+This ADR outlines the process we used to decide to sunset the NMDC Sandbox Submission Portal, the decision that was made, and the planned updated to ensure functionality is maintained. 
+
+## Context and Problem Statement
+
+Currently, the NMDC submission portal sandbox is wiped every Sunday.
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:13]: I elaborated on "wiped":

```suggestion
Currently, the NMDC Submission Portal sandbox is wiped every Sunday; i.e. all submissions are deleted from the underlying Postgres database.
```

File: decisions/0010-sunsetting-sandbox.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Some ambassadors or other users are still using the production environment for tests, workshops, and fake submissions. Additionally, people don't always know about the NMDC Sandbox site or there's a desire to have fake/test studies persist after a Sunday (ex: Prep for a workshop on Friday, and the submission still be there for Monday's session; Re-use the same submission for multiple workshops & not need to re-submit)
+
+We proposed a few solutions in the [GitHub dsicussion](https://github.com/microbiomedata/issues/discussions/743) and in a small meeting (contact Montana for those google notes). Team members were invited to contribute to and voice opinions on the GitHub discussion. Additionaly, we polled slack for reactions to confirm people were aware and we received 12 approvals.
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:16]: Fixed typos in "Discussion" and "Additionally":

```suggestion
We proposed a few solutions in the [GitHub Discussion](https://github.com/microbiomedata/issues/discussions/743) and in a small meeting (contact Montana for those Google notes). Team members were invited to contribute to and voice opinions via the GitHub Discussion. Additionally, we polled the team on Slack for reactions to confirm people were aware, and we received 12 approvals and 0 objections.
```

File: decisions/0010-sunsetting-sandbox.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Currently, the NMDC submission portal sandbox is wiped every Sunday.
+Some ambassadors or other users are still using the production environment for tests, workshops, and fake submissions. Additionally, people don't always know about the NMDC Sandbox site or there's a desire to have fake/test studies persist after a Sunday (ex: Prep for a workshop on Friday, and the submission still be there for Monday's session; Re-use the same submission for multiple workshops & not need to re-submit)
+
+We proposed a few solutions in the [GitHub dsicussion](https://github.com/microbiomedata/issues/discussions/743) and in a small meeting (contact Montana for those google notes). Team members were invited to contribute to and voice opinions on the GitHub discussion. Additionaly, we polled slack for reactions to confirm people were aware and we received 12 approvals. 
+
+<!-- This is an optional element. Feel free to remove. -->
+## Decision Drivers
+
+* People aren't using the sandbox site
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:21]: I assume some people have been using the sandbox environment (e.g. during workshops). 🤷 

I changed this to emphasize that the sandbox environment isn't housing _all_ test submissions.

```suggestion
* Even though we have been maintaining a sandbox environment, many people have continued to create test submissions in the production environment.
```

File: decisions/0010-sunsetting-sandbox.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Currently, the NMDC submission portal sandbox is wiped every Sunday.
+Some ambassadors or other users are still using the production environment for tests, workshops, and fake submissions. Additionally, people don't always know about the NMDC Sandbox site or there's a desire to have fake/test studies persist after a Sunday (ex: Prep for a workshop on Friday, and the submission still be there for Monday's session; Re-use the same submission for multiple workshops & not need to re-submit)
+
+We proposed a few solutions in the [GitHub dsicussion](https://github.com/microbiomedata/issues/discussions/743) and in a small meeting (contact Montana for those google notes). Team members were invited to contribute to and voice opinions on the GitHub discussion. Additionaly, we polled slack for reactions to confirm people were aware and we received 12 approvals. 
+
+<!-- This is an optional element. Feel free to remove. -->
+## Decision Drivers
+
+* People aren't using the sandbox site 
+* Sandbox is wiped and people may need test submissions to persist
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:22]:
```suggestion
* Sandbox is wiped weekly and people may need test submissions to persist longer than that
```

File: decisions/0010-sunsetting-sandbox.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+## Considered Options
+
+**Solution 1**
+* Provide a "delete this submission" button.
+    - Sandbox NEVER gets wiped. People can just delete their submissions when they're ready
+
+**Solution 2**
+* Extend the time.
+    - Submissions are deleted ever quarter?
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:33]: Fixed typo in "every":

```suggestion
    - Submissions are deleted every quarter?
```

File: decisions/0010-sunsetting-sandbox.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    - Submissions are deleted 90 days after creation?
+    - Should provide a warning? How?
+* Add a "trash" where people can restore the submission for so many days after deletion?
+
+**Solution 3**
+* Add a "delete this is a test" check box to the production site & those get deleted every so many days
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:39]:
```suggestion
* Add a "delete / this is a test" check box to the production site & those get deleted every so many days
```

File: decisions/0010-sunsetting-sandbox.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    - This checkbox can be checked after the fact so it's not deleted until someone says they want it to be.
+    - I think a LOT of people make test submission and forget about them, so the NMDC team should have the ability to delete them & send a notice
+
+## Decision Outcome
+
+Chosen option: Combo solution
+* Add “delete this submission” option to https://data.microbiomedata.org/submission/home
+    - Anyone with “creator role” (submitter & PIs) and admins can delete a submission
+* Eliminate sandbox
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:48]:
```suggestion
* Eliminate sandbox environment
```

File: decisions/0010-sunsetting-sandbox.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+* Add “Create test submission” button to [prod+dev](https://data.microbiomedata.org/submission/home)
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:49]:
```suggestion
* Add “Create test submission” button to all environments; e.g. [production](https://data.microbiomedata.org/submission/home), development, etc.
```

File: decisions/0010-sunsetting-sandbox.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    - Test, fake, workshop, TBD what we title this button
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:50]: I rephrased this to a form that I find easier to parse.

```suggestion
    - The button's label is TBD (e.g. "Test", "Fake", "Workshop", ...)
```

File: decisions/0010-sunsetting-sandbox.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    - Provide clear tool tip text to elaborate / explain when to make a new submission vs test submission
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:51]:
```suggestion
    - Provide clear tooltip text to elaborate / explain when to make a real submission vs test submission
```

File: decisions/0010-sunsetting-sandbox.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+        - Test submissions can NOT be submitted
+        - Test submissions can NOT be converted to real submissions
+        - Test is not equal to Draft
+* Eliminate deletion policy until we get a good sense of HOW test submissions are being created & IF people delete their own tests
+    - Submissions will NEVER be auto deleted
+* Add a “date last modified” attribute to submissions to track how often people refer to a test submission & determine when tests could be deleted
+* Add "Create test submission” option to the app
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:58]: Assuming you are referring to the NMDC Field Notes mobile app:

```suggestion
* Add "Create test submission” option to the NMDC Field Notes mobile app
```

File: decisions/0010-sunsetting-sandbox.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+* Add ability to track what submissions in the submission portal come from the app (for metrics)
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:59]:
```suggestion
* Add ability to track what submissions in the submission portal come from the mobile app (for metrics)
```

Comment (eecavanna) [reply decisions/0010-sunsetting-sandbox.md:59]: I think this was added already, via https://github.com/microbiomedata/nmdc-server/pull/1300 (CC: @pkalita-lbl)

File: decisions/0010-sunsetting-sandbox.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+<!-- This is an optional element. Feel free to remove. -->
+### Consequences    
+
+**Pros**:
+- dataportal sandbox is confusing, eliminates this
+- Users forget about / don't access sandbox
+    - One place makes it easier
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:67]:
```suggestion
- Data Portal sandbox environment will no longer exist
```

File: decisions/0010-sunsetting-sandbox.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+- Less code to manage
+
+**Cons**:
+- Need to update all documentation that references the -sandbox.
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:71]:
```suggestion
- Need to update all documentation that references the sandbox environment
```

File: decisions/0010-sunsetting-sandbox.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+## Summary
+
+This ADR outlines the decision and some goals of this change. The work is still in progress & this ADR will be updated when documentation is updated & sandbox is officially sunset. For additional details on discussions you can contact Montana or refer to the above linked GitHub issue.
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:75]:
```suggestion
This ADR outlines the decision and some goals of this change. The work is still in progress & this ADR will be updated when documentation is updated & the sandbox has been eliminated. For additional details on discussions, you can contact Montana or refer to the above linked GitHub issue. 
```

File: decisions/0010-sunsetting-sandbox.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
+---
+status: accepted
+date: 2024-08-14
+deciders: Montana Smith, NMDC PIs, Mike Nagler, Patrick Kalita 
+consulted: Entire NMDC team via [GitHub discussion](https://github.com/microbiomedata/issues/discussions/743), meetings, and slack were given the oppotrunity to voice opinions
+---
+# Sunsetting the NMDC Sandbox UI
+
+This ADR outlines the process we used to decide to sunset the NMDC Sandbox Submission Portal, the decision that was made, and the planned updated to ensure functionality is maintained.
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:9]:
```suggestion
This ADR outlines the process we used to decide to eliminate the NMDC Submission Portal sandbox environment, the decision that was made, and the planned updated to ensure functionality is maintained. 
```

File: decisions/0010-sunsetting-sandbox.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
+---
+status: accepted
+date: 2024-08-14
+deciders: Montana Smith, NMDC PIs, Mike Nagler, Patrick Kalita 
+consulted: Entire NMDC team via [GitHub discussion](https://github.com/microbiomedata/issues/discussions/743), meetings, and slack were given the oppotrunity to voice opinions
+---
+# Sunsetting the NMDC Sandbox UI
```
Comment (eecavanna) [inline decisions/0010-sunsetting-sandbox.md:7]:
```suggestion
# Eliminating the NMDC Submission Portal sandbox environment
```

**Comment by mslarae13** [comment]
File & all edits moved to https://github.com/microbiomedata/issues/pull/824

**Pull request closed without merging**